src/main/proto/marathon.proto

```diff
@@ -106,7 +126,8 @@ message MarathonTask {
   optional mesos.TaskStatus status = 9;
   optional mesos.SlaveID slaveId = 10;
   repeated mesos.NetworkInfo networks = 11;
-  optional ReservationWithVolumes reservation_with_volumes = 12; // since 0.16, a list of volumes can be associated with the task ID
+  optional Reservation reservation = 12; // since 0.16, a list of volumes can be associated with the task ID
+  optional ReservedState reserved_state = 13;
```

Comment:
Fair point – The `ReservedState` is only a means of expressing the state of a `Reserved` task.

The way it is modeled now, a resident task is either `LaunchedOnReservation`, or `Reserved` - the latter one can be in one of the states `New` (has never been launched), `Suspended` (the mesos task terminated), `Garbage` (unreserve/destroy, unused) and `Unknown` (unused; could be created if we receive unknown reservations/volumes and want to timeout before destroying them).

It's not easy to find a proper representation of the scala model here, but I'm open to discuss this!